daemon/c_vol.c

```diff
@@ -1053,6 +1107,47 @@ c_vol_start_child(c_vol_t *vol)
 		goto error;
 	}
 
+	//FIXME should be before mounting images, because ist sets up storage for bound files!
+	if (c_vol_do_shared_bind_mounts(vol) < 0) {
+		ERROR("Could not do shared bind mounts for container start");
+		goto error;
+	}
+	DEBUG("Mounting /dev");
+	const char *mount_data = is_selinux_enabled() ? "rootcontext=u:object_r:device:s0" : NULL;
+	const char *devfstype =
+		guestos_get_feature_devtmpfs(container_get_guestos(vol->container)) ? "devtmpfs" :
+										      "tmpfs";
+	unsigned long devopts = MS_RELATIME | MS_NOSUID;
+	char *dev_mnt = mem_printf("%s/%s", root, "dev");
+	int uid = container_get_uid(vol->container);
+	char *tmpfs_opts = (mount_data) ? mem_printf("uid=%d,gid=%d,%s", uid, uid, mount_data) :
+					  mem_printf("uid=%d,gid=%d", uid, uid);
+
+	if (mkdir(dev_mnt, 0755) < 0 && errno != EEXIST)
+		WARN_ERRNO("Could not mkdir %s", dev_mnt);
+
+	if (guestos_get_feature_devtmpfs(container_get_guestos(vol->container))) {
+		if (mount("/dev", dev_mnt, devfstype, devopts | MS_BIND, mount_data) < 0)
+			WARN_ERRNO("Could not bind mount /dev");
+
+		if (mount("/dev", dev_mnt, devfstype, devopts | MS_BIND | MS_RDONLY | MS_REMOUNT,
+			  mount_data) < 0)
+			WARN_ERRNO("Could not remount /dev (ro)");
+
+		//mount writable tmpfs over /dev
+		if (c_vol_mount_overlay(dev_mnt, "tmpfs", devfstype, devopts, tmpfs_opts, NULL,
+					NULL) < 0)
+			WARN_ERRNO("Could not mount tmpfs overlay to /dev");
```

Comment:
goto error?

Reply:
this is just moved code not really part of this commit. open an issue for this.